backend/siarnaq/api/episodes/admin.py

```diff
@@ -10,10 +16,49 @@
     Tournament,
     TournamentRound,
 )
+from siarnaq.api.user.models import User
+from siarnaq.gcloud import titan
 
 logger = structlog.get_logger(__name__)
 
 
+@admin.action(description="Export all submitted resumes")
+def export_resumes(modeladmin, request, queryset):
+    users = list(
+        User.objects.annotate(
+            rating=Max(
+                "teams__profile__rating__value", filter=Q(teams__episode__in=queryset)
+            )
+        )
+        .filter(profile__has_resume=True, rating__isnull=False)
+        .order_by("-rating")
+    )
+    rank_len = len(str(len({user.rating for user in users})))
+    with tempfile.SpooledTemporaryFile() as f:
+        with ZipFile(f, "w") as archive:
+            rank, last_rating = 0, None
+            for user in users:
+                if user.rating != last_rating:
+                    rank, last_rating = rank + 1, user.rating
+                rank_str = "rank-" + str(rank).zfill(rank_len)
+                user_str = user.first_name + "-" + user.last_name
+                if not user_str.isascii():
+                    user_str = "nonascii-user"
+                fname = f"{rank_str}-{user_str}.pdf"
```

Comment:
move fname creation into the if block. Also, consider moving the "check rating increment" into the if block too?